src/RepositoryChainsArchiver.ts

```diff
@@ -51,105 +63,107 @@ export default class RepositoryChainsArchiver {
-        }
-
-        if (matchType !== "full_match" && matchType !== "partial_match") {
-          console.log("Skipping non-match folder:", filePath);
-          continue;
+    try {
+      const uploadedFiles: { path: string; sizeInBytes: number }[] = [];
+      for (const matchType of ["full_match", "partial_match"]) {
+        for (const chainId of this.chainIds) {
+          for (let i = 0; i < 256; i++) {
+            const byte = i.toString(16).padStart(2, "0");
+            console.log(`Processing ${matchType} ${chainId} ${byte}`);
+            const pattern = `${matchType}/${chainId}/0x${byte}*/**/*`;
+            const entries = fg.stream(pattern, {
+              cwd: baseDir,
+              onlyFiles: true,
+              concurrency: READ_CONCURRENCY,
+              caseSensitiveMatch: false,
+            });
+
+            // Create tar stream for current byte
+            const archiveName = `${matchType}.${chainId}.${byte}.tar.gz`;
+            const localPath = path.join(this.exportPath, archiveName);
+            let currentTarStream: tar.Pack | undefined;
+
+            for await (const entry of entries) {
+              if (!currentTarStream) {
+                currentTarStream = createPack(localPath);
+              }
+
+              const relativePath = entry.toString();
+              const filePath = path.join(this.repositoryPath, relativePath);
+
+              const pathParts = relativePath.split("/");
+              const address = pathParts[2]; // Assuming path format: 'matchType/chainId/addressDirName/...'
+
+              if (!address.startsWith("0x")) {
+                console.log("Skipping non-address folder:", filePath);
+                continue;
+              }
+
+              await this.addFileToPack(currentTarStream, filePath);
+            }
+
+            if (currentTarStream) {
+              currentTarStream.finalize();
+
+              // Upload file
+              const s3Key = `${backupName}/${archiveName}`;
+              await this.uploadFile(localPath, s3Key);
+              uploadedFiles.push({
+                path: s3Key,
+                sizeInBytes: fs.statSync(localPath).size,
+              });
+
+              // Delete file
+              await fs.promises.rm(localPath, {
+                recursive: true,
+                force: true,
+              });
+              console.log(`Deleted local file: ${localPath}`);
+            }
+          }
         }
-
-        const firstTwoChars = address.slice(2, 4);
-        const firstByte = firstTwoChars.toUpperCase();
-
-        // Add the file to the appropriate tar stream
-        await this.addFileToPackStream(matchType, chainId, firstByte, filePath);
-      }
-
-      // After all files have been added, finalize the tar streams
-      for (const archiveName of Object.keys(this.tarStreams)) {
-        this.tarStreams[archiveName].finalize();
       }
+      this.uploadManifest(backupName, uploadedFiles);
+      this.deleteOldBackups();
```

Comment:
We need to await, no?

```suggestion
      await this.uploadManifest(backupName, uploadedFiles);
      await this.deleteOldBackups();
```

Reply:
good catch :+1: